tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py

```diff
@@ -72,6 +73,9 @@ def close_app(self, app):
         self.wait_for_card_ready(app)
         self.marionette.find_element(*self._close_button_locator(app)).tap()
         self.wait_for_element_not_present(*self._app_card_locator(app))
+        if len(self.cards) == 0:
+            self.marionette.switch_to_frame()
+            self.wait_for_condition(lambda m: self.apps.displayed_app.name == Homescreen.name)
```

Comment:
Otoh, there are lots of other methods that are not really doing that, for instance self.device.hold_home_button() where you have to use a bunch of stuff to make sure the cards_view is ready.